input/cql/IMMZDT14.cql

```diff
@@ -9,91 +9,80 @@ using FHIR version '4.0.1'
 context Patient
 
-/* 
- * @dataElement Typhoid vaccine immunization history
- */
-define "Typhoid vaccine immunization history":
-	0 // TODO: Define this
-
-/* 
- * @dataElement No-doses
- */
-define "No-doses":
-	0 // TODO: Define this
-
-/* 
- * @dataElement Age
- */
-define "Age":
-	0 // TODO: Define this
 
-/* 
- * @dataElement 2 years
+/*
+ * Rule: Should vaccinate patient with TCV vaccine on a 1 dose scheme
+ * Annotations:
+ * 	 - Provide Typhoid immunizations – using the "TCV vaccine immunization – NO PREVIOUS" schedule (1 dose scheme)
+ * Outputs:
+ * 	 - Immunize Patient for Typhoid -  No doses
+ * References:
+ * 	- WHO recommendations for routine immunization - summary tables: https://w ww.who.int/teams/immunization-vaccines-and-biologicals/policies/who-recommendations-for-routine-immunization---summary-tables
+ * Logic:
+ *	 ((((("Typhoid vaccine immunization history" = "No-doses") and ("Current Patient Age in Months" >= 6)) and ("Current Patient Age in Years" <= 45)) and ("Region recommends Typhoid vaccination" = true)) and ("Country is administering the TCV vaccine" = true))
  */
-define "2 years":
-	0 // TODO: Define this
+define "Should vaccinate patient with TCV vaccine on a 1 dose scheme":
+	IMMZCom."No Typhoid Doses Administered to Patient"
+	and IMMZCom."Current Patient Age In Months" >= 6
+	and IMMZCom."Current Patient Age In Years" <= 45 
+	and IMMZConf."Region recommends Typhoid vaccination"
+	and IMMZConf."Country is administering the TCV vaccine"
 
 /*
- * Rule: Should vaccinate patient with ViPS vaccine
+ * Rule: Should vaccinate patient with ViPS vaccine on a 1 dose scheme
  * Annotations:
  * 	 - Provide Typhoid immunizations – using the "ViPS vaccine immunization – NO PREVIOUS" schedule (1 dose scheme)
  * Outputs:
  * 	 - Immunize Patient for Typhoid - No Doses
  * References:
  * 	- WHO recommendations for routine immunization - summary tables: https://w ww.who.int/teams/immunization-vaccines-and-biologicals/policies/who-recommendations-for-routine-immunization---summary-tables
  * Logic:
- *	 (("Typhoid vaccine immunization history" = "No-doses") and ("Age" >= "2 years"))
+ *	 (((("Typhoid vaccine immunization history" = "No-doses") and ("Current Patient Age in Years" >= 2)) and ("Region recommends Typhoid vaccination" = true)) and ("Country is administering the ViPS vaccine" = true))
  */
-define "Should vaccinate patient with ViPS vaccine":
-	(("Typhoid vaccine immunization history" = "No-doses") and ("Age" >= "2 years"))
+define "Should vaccinate patient with ViPS vaccine on a 1 dose scheme":
+	IMMZCom."No Typhoid Doses Administered to Patient"
+	and IMMZCom."Current Patient Age In Years" >= 2 
+	and IMMZConf."Region recommends Typhoid vaccination"
+	and IMMZConf."Country is administering ViPS vaccine"
 
 /* 
- * @dataElement 6 years
+ * @dataElement Patient has completed Typhoid primary series
  */
-define "6 years":
-	0 // TODO: Define this
+define "Patient has completed Typhoid primary series":
```

Comment:
@mo3dev / @nityan - Verify this is accurate representation of the "completion of Typhoid" case